server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapperTests.java

```diff
@@ -755,6 +756,20 @@ public void testAddDocumentsToIndexBefore_V_7_5_0() throws Exception {
         assertArrayEquals("Decoded dense vector values is not equal to the indexed one.", validVector, decodedValues, 0.001f);
     }
 
+    public void testValidateOnBuild() {
+        final MapperBuilderContext context = MapperBuilderContext.root(false, false);
+
+        // Build a dense vector field mapper with float element type, which will trigger int8 HNSW index options
+        DenseVectorFieldMapper mapper = new DenseVectorFieldMapper.Builder("test", IndexVersion.current()).elementType(ElementType.FLOAT)
+            .build(context);
+
+        // Change the element type to byte, which is incompatible with int8 HNSW index options
+        DenseVectorFieldMapper.Builder builder = (DenseVectorFieldMapper.Builder) mapper.getMergeBuilder();
+        builder.elementType(ElementType.BYTE);
+        IllegalArgumentException e = expectThrows(IllegalArgumentException.class, () -> builder.build(context));
+        assertThat(e.getMessage(), containsString("[element_type] cannot be [byte] when using index type [int8_hnsw]"));
```

Comment:
Can we also test when the failure is thrown from the build method? That should happen if an invalid index option is set after modifying the element type?

Reply:
This test already covers when an `IllegalArgumentException` is thrown by the `build` method. Currently, there is no programmatic way to set the index options (only through parsing), so we can't change those like we do with the element type in this test. In fact, that's why the test is constructed the way it is; we need to first set the element type to `FLOAT` to create the desired default index options, then we change the element type to `BYTE` to create the incompatibility.

I thought about adding a way to programmatically set the index options, but that didn't seem appropriate when we would only use it in tests.